askui/askui
# feat: Interactive CLI

### Feature Description
Introduction of an interactive CLI where you can run `npx askui@latest init` and it asks as bunch of questions to set up AskUI in one go without the need to fumble with dependencies.

### What was Introduced

#### Functionality
The CLI asks about the following things which can also be directly given to the CLI with parameters:

- Platform (windows | macos | linux)
- Programming Language (Only TypeScript for now)
- Test Framework (jest | jasmine)
- Workspace ID
- Access Token
- Proxy Usage
- Creation of tsconfig

Depending on the answers it initializes an askui setup in the directory `askui_examples`, installs all necessary dependencies and populates the config files.

#### Docs

- The docs now contain the simplified `npm run askui` command everywhere instead of the `npx jest ...` one.
- The Getting Started Guides where updated, reflecting the different approaches for Windows, Linux and macOS

## How to Test

- Interactive CLI:
   `cd packages/askui-nodejs`
   `npm run build`
   Then you can execute the built version (see `dist` folder) by running the `bin/askui` executable in an empty folder for example like this:
   `./<path to bin>/askui`
- Docs: `cd docs` -> `npm run build` -> `npm run serve`
